docs/components/MessagingAttachmentView.jsx

```diff
@@ -0,0 +1,264 @@
+                  uploadError={attachment.uploadError}
+                  uploadComplete={attachment.uploadComplete}
+                />
+              ))}
+            </FlexBox>
+          </ExampleCode>
+        </Example>
+
+        {this._renderProps()}
+      </View>
+    );
+  }
+
+  // TODO: Update prop documentation.
+  // Add new props
+  _renderProps() {
+    return (
+      <>
+        <PropDocumentation
+          title="<MessagingAttachment /> Props"
+          availableProps={[
+            {
+              name: "attachmentID",
+              type: "string",
+              description: "Unique ID used to identify this attachment",
+              // optional: false,
+            },
+            {
+              name: "errorMsg",
+              type: "string",
+              description: "Optional error message, used when there is an error uploading, etc.",
+              optional: true,
+            },
+            {
+              name: "icon",
+              type: "React.ReactNode",
+              description:
+                "A ReactNode to use as the icon for the Attachment. Generally will be a call to <FileAttachmentIcon /> ",
+              // optional: false,
+            },
+            {
+              name: "onClickAttachment",
+              type: "(attachmentID: string) => void",
+              description: "Callback to be called upon clicking the Attachment",
+              // optional: false,
+            },
+            {
+              name: "onRemoveAttachment",
+              type: "(attachmentID: string) => void",
+              description: "Callback to be called upon clicking the Remove Attachment (X) button",
+              optional: true,
+            },
+            {
+              name: "title",
+              type: "string",
+              description: "Title of the file",
+              // optional: false,
+            },
+            {
+              name: "subtitle",
+              type: "string",
+              description: "Optional subtitle, generally used for upload status, filesize, etc",
+              optional: true,
+            },
+            {
+              name: "isUpload",
+              type: "boolean",
+              description:
+                "Optional boolean for whether this Attachment is a draft/upload or it is a sent Attachment",
+              defaultValue: "false",
+              optional: true,
+            },
+            {
+              name: "uploadComplete",
+              type: "boolean",
+              description: "Optional boolean for whether this Attachment has finished uploading",
+              defaultValue: "false",
+              optional: true,
+            },
+            {
+              name: "uploadError",
+              type: "boolean",
+              description:
+                "Optional boolean for whether there was an error while uploading the Attachment",
+              defaultValue: "false",
+              optional: true,
+            },
```

Comment:
This totally works, but what's the advantage over the `errorMsg?: string` approach mentioned in the spec, where we use something like `!!errorMsg` as the boolean check for if the component is in an error state, and `errorMsg` itself is the error message we display, rather than it being fed in through `title`?

Seems like having `errorMsg` means we don't need `uploadError`, basically.